### PR TITLE
add graphql endpoint

### PR DESCRIPTION
**Pull request by dpopp07**
Added a GraphQL endpoint that can return a list of `Drink` objects and whatever data you ask for from the drinks. I used the standard `graphql-express` library from GraphQL but may switch to an Apollo server in the future depending on needs. I like their stuff.

I structured the organization in a way I thought was modular and efficient. After re-working it a few times to get it how I liked it, I realized it pretty much follows the structure of [this learning project](https://github.com/FrontendMasters/api-design-node-v2) which I looked at to begin with but thought the structure was gross. So after working with it a bunch, I now understand why they organized it that way. The structure can evolve over time but I think this is a good start.

It's not ready yet (`in progress`) but I wanted you to be able to start looking at it so you can ask any questions, make suggestions, etc. You can use [the explorer](http://localhost:3001/api/v2/graphql) to play around with it, but you will need to comment out *Line 27* in `server.js` because the explorer doesn't work with authentication. Working on a fix for this, it's a hot topic in the community because the `graphiql` people are basically just telling people to find a complex workaround.

**TODO**
- [x] Add functionality for adding Users, so you can actually integrate with it
- [x] Add tests for user mutations
- [x] Better organization for the tests
- [x] Get the `graphiql` explorer accessible with authentication for development

*Note* Testing is weird. I'm not too sure about a good way to write unit tests yet. For that reason, the unit test coverage has significantly dropped, but the actual testing coverage will go up a good amount. If you have any ideas, feel free to hit me up with them.

### CODE REVIEW

**Comment by dpopp07** [comment]
Currently, the best way to authenticate GraphIQL requests is to use [this standalone app](https://github.com/skevy/graphiql-app). It is exactly `graphiql` but with the ability to include headers. It is sort of like a lightweight Postman for GraphQL. We will have to use it for development until something better comes around but I'm marking this as resolving that sub-issue.

(Or commenting out `app.use(checkJwt);` in `server.js`, which is what I have been doing)

**Comment by dpopp07** [comment]
Okay, I finished it up. For the test, I decided to organize them with nested `describe`'s (common practice). During this, I realized that tests to cover the error handler are endpoint-agnostic and so I decided to make a single test suite purely for covering these types of errors, rather than run these tests for every single resource unnecessarily. This led to a good amount of refactoring and reorganizing and re-optimizing (again 😂, and probably won't be the last time). So, this ended up being a large PR. Sorry about that.

I made sure to leave the `POST User` structure the way it was before! I will be changing that in a separate PR.

File: bartop-api/test/utils/testObjects/user/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const user0 = {
+  id: 'marvin@gmail.com|7984284578419749'
+};
+
+const user1 = {
+  id: 'megan@yahooanswers.com|1988284318489743'
+};
+
+const user2 = {
+  id: 'bilbo@bartopbadboi.io|5467945134679452'
```
Comment (jamesvclements) [inline bartop-api/test/utils/testObjects/user/index.js:10]: 💀 💀

**Comment by jamesvclements** [comment]
Looking goooood. Could you add something to the README about hitting http://localhost:3001/api/v2/graphql to view the interactive explorer?

File: README.md [hunk 2]
```diff
@@ -75,7 +75,13 @@ To create a new database use the [RethinkDB dashboard](http://localhost:8080/#ta
...
 First, make sure the database is running. Then, start the API server:
 `npm start`
-The API will be available for requests at http://localhost:3001 (if you set BARTOP_API_PORT to 3001 in your `.env` file).
+The API will be available for requests at http://localhost:3001 (if you set BARTOP_API_PORT to 3001 in your `.env` file, which is recommended). Requests to the API in development do not need to be authorized. For testing and production, the API requires authentication/authorization using JSON Web Tokens.
+
+##### Explore the v1 REST API
+The REST API is available under the route `/api/v1`. Use a tool like [Postman](https://www.getpostman.com/apps) to test different endpoints. Documentation coming soon...
+
+##### Explore the v2 GraphQL API
+The v2 GraphQL API is available under the single endpoint `/api/v2/graphql`. The recommended way to explore this API is using the GraphIQL tool, available by visiting [the above route](http://localhost:3001/api/v2/graphql) in a browser. Here, there is an environment to test queries alongside a panel with documentation. If desired, the v2 GraphQL API can also be explored using a tool like Postman.
```
Comment (jamesvclements) [inline README.md:84]: 👏

**Review by jamesvclements — APPROVED** [review]
Well done sir

**Pull request merged**